tethysapp/temp_precip_trends/functions.py

```diff
@@ -0,0 +1,192 @@
+    time_series['cum_pr_mm'] = cum_precip
+
+    return jsonify(time_series, 'cum_pr_mm')
+
+
+def jsonify(dataset, variable):
+    """
+    Extract a time series from a THREDDS dataset at the given location.
+
+    Args:
+        dataset(xarray.Dataset): an xarray dataset.
+        variable(str): Name of the variable to query.
+
+    Returns:
+        Python Dict: JSON-compatible Python Dict.
+    """
+    df = pd.DataFrame(data={variable: np.transpose(dataset[variable].data)}, index=dataset.time.data)
+    df.index = df.index.strftime('%Y-%m-%dT%H:%M:%SZ')
+    df.index.name = 'datetime'
+
+    context = {
+        'time_series': {
+            'datetime': df.index.tolist(),
+            variable: df[variable].to_list()
+        }
+    }
+
+    context['time_series'].update(df.to_dict(orient='list'))
+
+    return context
+
+
+def param_check(request):
+    """
+    Check that required parameters were passed.
+
+    Args:
+        request(Django HttpRequest): Django GET request with input parameters.
+
+    Returns:
+        Python Dict: Dictionary with success or error key  and a message.
+    """
+    if request.method != "GET":
+        return {"error": "only GET requests are allowed."}
+    elif not request.GET.get("geometry", None):
+        return {"error": "'geometry' is a required parameter."}
+    elif not request.GET.get("end_time", None):
+        return {"error": "'end_time' (YYYYMMDD) is a required parameter."}
+    else:
+        return {"success": "required parameters provided."}
+
+
+def overlap_ts(time_series):
+    """
+    Updates time series by adding one year to time-series dates
+
+    Args:
+        time_series(Python Dict): times_series dict returned by the get_data or get_cum_precip_data functions.
+
+    Returns:
+
+    """
+    new_date_list = []
+    for date in time_series['time_series']['datetime']:
+        new_date = f'{date[:3]}{int(date[3]) + 1}{date[4:]}'  # add one to the year for projected data overlap
+        new_date_list.append(new_date)
+
+        time_series['time_series']['datetime'] = new_date_list
+
+
+def extract_time_series_at_location(catalog, geometry, variable, start_time=None, end_time=None, vertical_level=None):
+    """
+    Extract a time series from a THREDDS dataset at the given location.
+
+    Args:
+        catalog(siphon.catalog.TDSCatalog): a Siphon catalog object bound to a valid THREDDS service.
+        geometry(geojson): A geojson object representing the location.
+        variable(str): Name of the variable to query.
+        start_time(datetime): Start of time range to query. Defaults to 9 months before end_time.
+        end_time(datetime): End of time range to query. Defaults to datetime.utcnow().
+        vertical_level(number): The vertical level to query. Defaults to 100000.
+
+    Returns:
+        xarray.Dataset: The data from the NCSS query.
+    """
+    try:
+        dataset = catalog.datasets['ERA5 Daily Precipitation and Temperatures']
```

Comment:
Put the name of the dataset in an app variable and have it default to this value.

Reply:
done. api helper functions now take a dataset as param instead of a catalog, and the ds name is in app.py